src/components/auth-provider.tsx

```diff
@@ -51,9 +52,11 @@ const AuthProvider: React.FC< AuthProviderProps > = ( { children } ) => {
 		}
 
 		const { token } = payload;
+		const newClient = createWpcomClient( token.accessToken, locale );
 
 		setIsAuthenticated( true );
-		setClient( createWpcomClient( token.accessToken, locale ) );
+		setClient( newClient );
+		setWpcomClient( newClient );
```

Comment:
Would it be possible and make sense to use only one client set?
I've noticed some duplication in the code when we set or unset it.

```ts
setClient( undefined );
setWpcomClient( undefined );
```